src/server/req_runjob.c

```diff
@@ -771,7 +770,11 @@ req_runjob2(struct batch_request *preq, job *pjob)
 		/* provisioning was needed and was qneueued successfully */
 		/* Allways send ack for prov jobs, even if not async run */
 		reply_ack(preq);
-		return;
+		if (save_subjob(pjob)) {
+			free_nodes(pjob);
+			req_reject(PBSE_SAVE_ERR, 0, preq);
```

Comment:
a req_reject() after a reply_ack() is not meaningful. I suggest we move the reply_ack() to after this block

Reply:
Don't all of the reply/reject functions free the batch reply structure?  Won't doing 2 of them double free?